### ISSUE TITLE
seaborn.objecs.Plot.save does not include legend

### ISSUE THREAD

**Issue by evanhorn** [mwaskom/seaborn#3057]
I am very excited about the new seaborn.objects interface. However, when using it and save a plot to a png file, the legend is omitted, but it is shown when plotting to the screen. (See below — first image is a screen capture, second image is the same image saved as a file.)

![test](https://user-images.githubusercontent.com/5512612/194420685-434786fe-7a77-4749-88d4-b85024ec851a.png)

![perf_plate_amplitude_test](https://user-images.githubusercontent.com/5512612/194420715-fa38c2dc-7e65-4672-a98e-c7a90abf5623.png)

**Comment by mwaskom** [comment]
Yeah this is a rough edge. The easiest thing to do would be to delegate this to matplotlib, but their auto layout engines can't handle a figure legend. There's some code in `FacetGrid` to expand the figure size and then shrink the subplots to fit the legend, but it's pretty brittle and I wanted to avoid bringing it over to the new interface unless necessary. So this was left without a clear solution to avoid getting stuck committing to a suboptimal approach.

What's happening when you show the plot in a notebook is that it's saving the figure with `bbox_inches="tight"`, which expands the figure so that all elements fit. The downside to this is that the figure size ends up different (larger *or* smaller) than what you might specify in `theme` or `layout`. But if that's ok, it will work in `Plot.save` too.

Alternatively, the legend is not placed outside of the plot by default when you use `Plot.on` because it expects that there might be other subplots / subfigures  in that space, and so that it will work well in a GUI context. So (as you discovered) that's another temporary workaround.

**Comment by gfairbro** [comment]
I just thought i would add that actually the legend is there, but it is being cut off (apologies if this is obvious). you can just barely see the edge above in the saved file. It's more obvious in my example: 

![Legend_Direction](https://user-images.githubusercontent.com/88786321/196810526-6ee18ca3-0f9a-4211-a37e-d675d54c8784.jpg)

**Comment by jalsti** [comment]
This seems to be related to #3130. When the plot gets created with pyplot during `show()`, the legend location gets set fixed to "center left", which is above the plot's graph area. During `save()` pyplot is not used, which results in the hardcoded position "center right", which is more or less on the outside because of the hardcoded `bbox_to_anchor` for the legend. With the PR #3128 related to the issue the legend could at least be set explicitly.

**Comment by leodarrigade** [comment]
I could solve this issue by passing `bbox_inches='tight'` to the save function. It worked when I used the Seaborn save function as
```python
so.Plot(...).save('test.png', bbox_inches='tight)
```
or when I used Matplotlib:
```python
f = plt.figure()
p.on(f)
f.savefig("test.png", bbox_inches='tight)
```
Note that in the latter case, if I did not specified `bbox_inches='tight'` to `savefig`, the legend was cutoff (contrary to what @evanhorn said).

**Comment by madSec** [comment]
Hi,
Just found that when showing a plot rather than saving one can use the "legend" method with the "engine" argument as shown below to move the legend into the frame of the plot.

fig = plt.figure()
(
so.Plot()
.add(so.Dot())
.layout(engine="tight")
.on(fig)
.show()
)

Strangely this only affects showing the image in the notebook for example, when saving the image I still need to add "bbox_inches='tight'" to the save methode, and even then in the saved image the legend is located outside the frame.
